lib/oauth2/client.ex

```diff
@@ -202,6 +207,43 @@ defmodule OAuth2.Client do
     url
   end
 
+  @doc """
+  Register a serialization module for a given mime type.
+
+  ## Example
+
+      iex> client = OAuth2.Client.put_serializer(%OAuth2.Client{}, "application/json", Jason)
+      %OAuth2.Client{serializers: %{"application/json" => Jason}}
+      iex> OAuth2.Client.get_serializer(client, "application/json")
+      Jason
+  """
+  @spec put_serializer(t, binary, atom) :: t
+  def put_serializer(%Client{serializers: serializers} = client, mime, module)
+    when is_binary(mime) and is_atom(module) do
+    %Client{client | serializers: Map.put(serializers, mime, module)}
+  end
+
+  @doc """
+  Un-register a serialization module for a given mime type.
+
+  ## Example
+
+      iex> client = OAuth2.Client.delete_serializer(%OAuth2.Client{}, "application/json")
+      %OAuth2.Client{}
+      iex> OAuth2.Client.get_serializer(client, "application/json")
+      OAuth2.Serializer.Null
```

Comment:
I'm not sure how I feel about `OAuth2.Serializer.Null`, why would you not return `nil`?

Reply:
I think this is a great point. I think this is left over from a previous idea and I didn't take the time to remove it. 👍